acinclude/squid-util.m4

```diff
@@ -211,7 +211,7 @@ unset squid_tmp_define
 dnl aborts with an error specified as the second argument if the first argument doesn't
 dnl contain either "yes" or "no"
 AC_DEFUN([SQUID_YESNO],[
-  AS_IF([test "$1" != "yes" -a "$1" != "no"],[AC_MSG_ERROR([$2])])
+  AS_IF([test "$1" != "yes" -a "$1" != "no"],[AC_MSG_ERROR([Bad argument for $2, which does not accept a value.])])
```

Comment:
```
Bad argument for --enable-ecap, which does not accept a value.
```

The new error text looks wrong to me because `--enable-foo` options do accept at least two values: "yes" and "no". I realize that some of the existing callers had similar problems, but we should not hard-code those selected caller bugs into the macro implementation, of course.

Here is an untested correction sketch:

```suggestion
  AS_IF([test "$1" != "yes" -a "$1" != "no"],[AC_MSG_ERROR([Bad argument for $2: "$1". Expecting "yes", "no", or no argument.])])
```

As always, please adjust as needed to address the _essence_ of this change request. No need to argue about any specific suggestion flaws.